lib/cable_ready/cable_car.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "thread/local"
+require_relative "applicable"
+
+module CableReady
+  class CableCar
```

Comment:
There's clearly enough duplication and shared behavior/purpose in this class and `CableReady::Channel` that would afford extracting a common (abstract) base class?

Reply:
That's what I was trying to do with Applicable! I am not sure how I would cut deeper, but we're here to challenge each other. What do you have in mind?

Reply:
yes, I just noticed that large parts of `initialize`, as well as `reset`, and `transmittable` (all but for the name) are identical. That's why I personally would go with an abstract base class.

Reply:
That said, they don't really have a `is_a?` relationship. I think `Applicable` is fine, but I'd clearly extract more.

Reply:
I'm intrigued by the notion of a common base class. I was concerned because one is a Singleton and one isn't, but I think this is a red herring - it should be fine to make one of the subclasses a Singleton.

I'll give it a shot!

Reply:
https://github.com/leastbad/cable_ready/pull/3